docs/grid-introducer.rst

```diff
@@ -0,0 +1,104 @@
+
+The Grid Introducer is configured in ``tahoe.cfg`` in the ``[client]`` section's ``grid-introducer.cap`` and ``grid-introducer.furl`` items.
+For example::
+
+  [client]
+  grid-introducer.cap = URI:DIR2-RO:4bnx...
+  grid-introducer.furl = pb://sokl...@192.168.69.247:44801/eqpw...
+
+Start a Tahoe-LAFS client node with these items configured and the client will be able to find and follow all storage servers that are part of that grid.
+
+Use As A Storage Provider
+-------------------------
+
+You want your Tahoe-LAFS storage node to be able to publish its connection details where clients can find it.
+Using Grid Introducer,
+a storage server will discover other storage servers and write its connection details to all of them.
+Then the capability to read this information is linked into a collection stored on the grid and known to clients.
+
+A storage server must be configured just like a storage client so it can discover other storage servers.
+See `Use As A Client`_ for this part of the configuration.
+
+Next, a storage server must be configured to maintain its announcement on the grid.
+This is also done in ``tahoe.cfg``,
+in the ``[storage]`` section's ``grid-introducer.enable`` item.
+
+For example::
+
+  [storage]
+  grid-introducer.enable = true
+
+After the storage server has been configured this way and started it will upload its announcement.
+
+Use As A Grid Coordinator
+-------------------------
+
+Setup
+~~~~~
+
+You want to offer a collection of storage servers as a Tahoe-LAFS storage grid.
+
+The ``grid-introducer`` tool requires a client node it can use to store and publish announcements.
+With that in place,
+the first step is to create the introducer's persistent state::
+
+  grid-introducer create --config <path> --client-api-root <Tahoe-LAFS client node HTTP API root>
+
+This is an error if ``<path>`` exists already.
+If it does not then a new grid introducer configuration is written there.
+If ``<path>`` is ``-`` then the configuration is written to stdout.
+
+This is the grid introducer's state and is required by future commands.
+It is the operator's responsibility to persist this state.
+
+Enrolling Servers
+~~~~~~~~~~~~~~~~~
+
+A Tahoe-LAFS storage server which is to be enrolled first shares its announcement readcap with you.
+Then, you will add it to the announcement directory::
+
+   grid-introducer add-storage-server --config <path> --readcap URI:CHK-RO:5cmy...
+
+``<path>`` should have previously been created by ``grid-introducer create``.
+If ``<path>`` is ``-`` then the configuration is read from stdin.
+
+
+Inviting Clients
+~~~~~~~~~~~~~~~~
+
+A Tahoe-LAFS client node which is to use the grid introducer needs to be configured with a couple items.
+A configuration blob for clients can be generated like this::
+
+  grid-introducer generate-client-config --config <path>
```

Comment:
Following my idea from above, this could be `generate-invite` or similar instead and then the user doesn't need to copy-paste individual items out of a file, they simply inject the invite into their client's `tahoe join-grid` or similar command. For local use this could just use unix pipes (`tahoe grid-introducer --config testgrid/introducer invite-client | tahoe --node-directory testgrid/alice join-grid --invite -` or so).